e2e/app/admin_staking.go

```diff
@@ -0,0 +1,165 @@
+package app
+
+import (
+	"context"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/halo/genutil/evm/predeploys"
+	"github.com/omni-network/omni/lib/contracts/staking"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+var stakingAdminABI = mustGetABI(bindings.StakingAdminMetaData)
+
+// UpgradeStaking upgrades the staking predeploy.
+func UpgradeStaking(ctx context.Context, def Definition) error {
+	s, err := setup(def)
+	if err != nil {
+		return err
+	}
+
+	c, err := setupChain(ctx, s, chainOmniEVM)
+	if err != nil {
+		return err
+	}
+
+	return upgradeStaking(ctx, s, c)
+}
+
+// ConfigureStaking ensures on chain Staking config matches config in lib/config/staking.
+func ConfigureStaking(ctx context.Context, def Definition) error {
+	s, err := setup(def)
+	if err != nil {
+		return err
+	}
+
+	c, err := setupChain(ctx, s, chainOmniEVM)
+	if err != nil {
+		return err
+	}
+
+	return configureStaking(ctx, s, c)
+}
+
+func upgradeStaking(ctx context.Context, s shared, c chain) error {
+	calldata, err := stakingAdminABI.Pack("upgrade", s.admin, s.deployer)
+	if err != nil {
+		return errors.Wrap(err, "pack calldata")
+	}
+
+	out, err := runForge(ctx, scriptStakingAdmin, c.rpc, calldata, s.admin, s.deployer)
+	if err != nil {
+		return errors.Wrap(err, "run forge", "out", out)
+	}
+
+	log.Info(ctx, "Staking predeploy upgraded", "out", out)
+
+	return nil
+}
+
+func configureStaking(ctx context.Context, s shared, c chain) error {
+	client, err := ethclient.Dial(c.rpc)
+	if err != nil {
+		return errors.Wrap(err, "dial rpc")
+	}
+
+	contract, err := bindings.NewStaking(common.HexToAddress(predeploys.Staking), client)
+	if err != nil {
+		return errors.Wrap(err, "new staking contract")
+	}
+
+	cfg, ok := staking.ConfigByNetwork(s.network.ID)
+	if !ok {
+		log.Info(ctx, "No static staking config, skipping", "network", s.network.ID)
+		return nil
+	}
```

Comment:
Right now just logging and returning nil. Though we can configure these for devnet / staging & use them in our e2e flow. Right now just do it off of [genesis alloc accounts](https://github.com/omni-network/omni/blob/main/e2e/app/portalregistry.go#L265).

Another thing I need to add is how to tests - devnet will just deploy what's in binary. And this command uses what's in binary. So need to add an override for tests.